eth2util/keystore/keystore.go

```diff
@@ -121,6 +129,40 @@ func LoadKeys(dir string) ([]tbls.PrivateKey, error) {
 	return resp, nil
 }
 
+// orderByKeystoreNum orders keystore file names by their index in ascending order.
+func orderByKeystoreNum(files []string) ([]string, error) {
+	prefix := filepath.Dir(files[0])
+
+	extractor := regexp.MustCompile(`keystore-(?:insecure-)?([0-9]+).json`)
+
+	var err error
+	sort.Slice(files, func(i, j int) bool {
+		first := strings.TrimPrefix(files[i], prefix)
+		second := strings.TrimPrefix(files[j], prefix)
+
+		if !extractor.MatchString(first) || !extractor.MatchString(second) {
+			err = errors.New("keystore filenames do not match expected pattern")
+			return false
+		}
+
+		firstNumRaw := extractor.FindStringSubmatch(first)[1]
+		secondNumRaw := extractor.FindStringSubmatch(second)[1]
+
+		// PSA: we're avoiding error checking here because the regexp is explicitly expecting
+		// numbers in the first glob, hence Atoi will never fail.
+		firstNum, _ := strconv.Atoi(firstNumRaw)
```

Comment:
what is the cost of adding an error check here? seems trivial to add and doesn't complicate things much, I'd just add it, then regexp can't bite us later.

Reply:
The cost is insignificant, I can add it.

Although, I would trust the regexp engine to be right... but can *I* be trusted? We'll never know.

if err it is.